radar_ego_velocity_estimation_ros_node:
  ros__parameters:
    ## radar ego velocity estimation
    # filtering
    min_dist: 0.25                        # min distance of valid detection
    max_dist: 20.0                       # Max distance of valid detection
    min_db: 5.0                           # min SNR in [db]
    elevation_thresh_deg: 60.0            # threshold for elevation [deg]
    azimuth_thresh_deg: 60.0              # threshold fo azimuth [deg]
    filter_min_z: -100.0
    filter_max_z: 100.0
    doppler_velocity_correction_factor: 1.0 # Doppler velocity correction

    # zero velocity detection
    thresh_zero_velocity: 0.05        # all inliers need to smaller than this value
    allowed_outlier_percentage: 0.25  # outlier ratio (=percentage of detections which are allowed to be above thresh_zero_velocity)
    sigma_zero_velocity_x: 0.025      # sigma v_r
    sigma_zero_velocity_y: 0.025      # sigma_v_r
    sigma_zero_velocity_z: 0.025      # sigma v_r

    # noise offset
    sigma_offset_radar_x: 0.05   # offset added to estimated sigmas
    sigma_offset_radar_y: 0.025
    sigma_offset_radar_z: 0.05

    # result filtering
    max_sigma_x: 0.2   # max estimated sigma to be considered an inlier (right)
    max_sigma_y: 0.15  # (forward)
    max_sigma_z: 0.2   # (up)
    max_r_cond: 1.0e3  # max conditional number of LSQ Pseudo Inverse to ensure a stable result
    use_cholesky_instead_of_bdcsvd: True # faster but less stable

    # RANSAC parameters
    use_ransac: True     # turn on RANSAC LSQ
    outlier_prob: 0.4    # worst case outlier probability
    success_prob: 0.9999 # probability of successful determination of inliers
    N_ransac_points: 3   # number of measurements used for the RANSAC solution
    inlier_thresh: 0.15  # inlier threshold for inlier determination

    # ODR refinement
    use_odr: False    # turn on odr refinement
    sigma_v_r: 0.125  # noise of v_r measurement used for the refinement
    min_speed_odr: 4.0 # min speed for ODR refinement
    model_noise_offset_deg: 2.0 # min model noise
    model_noise_scale_deg: 10.0 # scale model noise

    run_without_trigger: True
